web/modules/custom/nys_senators/src/Plugin/Block/SenatorMicrositeMenu.php

```diff
@@ -31,49 +31,59 @@ class SenatorMicrositeMenu extends BlockBase implements ContainerFactoryPluginIn
    *
    * @var \Drupal\Core\Cache\CacheBackendInterface
    */
-  protected $cache;
+  protected CacheBackendInterface $cache;
 
   /**
    * The Entity TypeManager Interfacee.
    *
    * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
-  protected $entityTypeManager;
-
+  protected EntityTypeManagerInterface $entityTypeManager;
 
   /**
    * Current Route Match.
    *
    * @var \Drupal\Core\Routing\CurrentRouteMatch
    */
-  protected $routeMatch;
+  protected CurrentRouteMatch $routeMatch;
+
+  /**
+   * Current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected AccountProxyInterface $currentUser;
 
   /**
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
     return new static(
-          $configuration, $plugin_id, $plugin_definition,
-          $container->get('cache.default'),
-          $container->get('entity_type.manager'),
-          $container->get('current_route_match')
-      );
+      $configuration, $plugin_id, $plugin_definition,
+      $container->get('cache.default'),
+      $container->get('entity_type.manager'),
+      $container->get('current_route_match'),
+      $container->get('current_user'),
+    );
   }
 
   /**
    * Constructor.
    */
-  public function __construct(array $configuration,
-        $plugin_id,
-        $plugin_definition,
-        CacheBackendInterface $cache_backend,
-        EntityTypeManagerInterface $entity_type_manager,
-        CurrentRouteMatch $route_match
-    ) {
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    CacheBackendInterface $cache_backend,
+    EntityTypeManagerInterface $entity_type_manager,
+    CurrentRouteMatch $route_match,
+    AccountProxyInterface $current_user,
+  ) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
     $this->cache = $cache_backend;
     $this->entityTypeManager = $entity_type_manager;
     $this->routeMatch = $route_match;
+    $this->currentUser = $current_user;
```

Comment:
Consider `nys_users\UsersHelper::resolveUser()` instead of injecting the account proxy.

Reply:
See my comment in Jira: https://atendesign.atlassian.net/browse/NYS-109?focusedCommentId=197970 :)